rm: Add a data retention policy doc (#403)
foundriesio/docs: source/reference-manual/factory/data-retention.rst

.. _ref-data-retention:

Data Retention Policies
=======================

> Users often need to understand data retention policies within a FoundriesFactory so that they can work through compliance processes for their product.

Author: how about "your product" instead of "a product"?

ota-lite.foundries.io    Device gateway      GCP - us-central1-a
ostree.foundries.io      OSTree for devices  GCP - us-central1-a
======================   ================== =================================

These services can be grouped into two areas of concern: customer data and device data.

Customer Data
-------------

source.foundries.io
~~~~~~~~~~~~~~~~~~~
User source code like ``containers.git`` lives in Digital Ocean's SFO2 data center.
This data is periodically backed up within this data center.
Backups over 6 months old are pruned.
All customer source code is deleted when a Factory is deleted.

ci.foundries.io
~~~~~~~~~~~~~~~
ci.foundries.io runs inside Google's GCP us-central1-a region.
It stores data in two places:

 * Galera Cluster Database
 * Google Storage Bucket

The storage bucket is multi-regional in the United States.
The database is periodically backed up to Google Storage.
Backups over 6 months old are pruned.

CI Workers
~~~~~~~~~~
CI work takes place in ephemeral instances (Docker containers) that are removed upon the completion of a CI Run.
The exception to this is the `sstate cache`_ for LmP builds.

> This is kept on an NFS drive in the customer's CI region.

Author: its multi-tenant. we group customers into clusters that share a drive

> This data is periodically pruned to remove old data and completely deleted when a Factory is deleted.

Author: no. it depends on the rate of change within factories and how fast their data is becoming unused.

api.foundries.io
~~~~~~~~~~~~~~~~
This service is a reverse proxy to other services and stores no data.

app.foundries.io
~~~~~~~~~~~~~~~~
This service has two components:

 * A web view to api.foundries.io
 * Factory user and subscription management

The service runs inside Google's GCP us-central1-a region.
The user and subscription database is backed up nightly to Google Storage.
Backups over 6 months old are pruned.

Device Data
-----------
Foundries.io does not store any other device application data beyond what is listed in this section.

hub.foundries.io
~~~~~~~~~~~~~~~~
User container images are managed by this service.
Devices pull container updates from this service.
All data is stored in a multi-regional Google Storage Bucket in the United States.
It is deleted when a Factory is deleted.

ota-lite.foundries.io
~~~~~~~~~~~~~~~~~~~~~
This service manages two pieces of data:

 * The Factory's TUF Metadata
 * Device data

Device data includes things such as:

 * Hardware information from the lshw_ tool.
 * Details of the last 10 OTAs (``fioctl updates show``)
 * When the device was created
 * The Factory user that created the device

>  * The device's local IPv4 address. *Depends on deployment details, this is normally a class C IP Address and not the public IPv4 address it accesses the Internet from*

Author: i'll commit it and fix it if its not rendering well

 * The device's MAC address

> The service and data is inside Google's GCP us-central1-a region.

Reviewer: ```suggestion
Both service and data gets kept inside Google's GCP us-central1-a region.
```

The data is periodically backed up to Google Storage.

> Backups over 6 months old are pruned.

Reviewer: ```suggestion
Backups over 6 months old get pruned.
```
